crates/flox/src/commands/channel.rs

```diff
@@ -107,47 +108,35 @@ fn construct_search_params(search_term: &str, flox: &Flox) -> Result<SearchParam
 
     // We've already checked that the search term is Some(_)
     let query = Query::from_str(search_term)?;
-
-    Ok(SearchParams {
+    let params = SearchParams {
         registry,
         query,
         systems: Some(vec![flox.system.clone()]),
         ..SearchParams::default()
-    })
+    };
+    debug!("search params: {:?}", params);
+    Ok(params)
 }
 
 /// This function is a hack to convert the current subscriptions into a format
 /// that matches the search spec, which expects sources to come from the manifest.
 ///
 /// This is temporary and will be removed once we have a functioning manifest.
-fn collect_manifest_inputs(flox: &Flox) -> (HashMap<String, RegistryInput>, Vec<String>) {
-    let channels = flox
-        .channels
-        .iter()
-        .filter_map(|entry| {
-            if HIDDEN_CHANNELS.contains_key(&*entry.from.id) {
-                None
-            } else if DEFAULT_CHANNELS.contains_key(&*entry.from.id) {
-                Some((ChannelType::Flox, entry))
-            } else {
-                Some((ChannelType::User, entry))
-            }
-        })
-        .sorted();
-
-    // Create `registry` parameter for `pkgdb`
-    let mut priority: Vec<String> = Vec::new();
-    let mut inputs = HashMap::new();
-    for (_, entry) in channels {
-        priority.push(entry.from.id.to_string());
-        let input = RegistryInput {
-            from: entry.to.clone(),
-            // TODO: handle `subtrees` and `stabilities`
-            subtrees: None,
-            stabilities: None,
-        };
-        inputs.insert(entry.from.id.to_string(), input);
-    }
+fn collect_manifest_inputs(_flox: &Flox) -> (HashMap<String, RegistryInput>, Vec<String>) {
+    let priority = vec!["nixpkgs".to_string()];
+    let nixpkgs_json = json!({
+        "type": "github",
+        "owner": "NixOS",
+        "repo": "nixpkgs",
+        "rev": "e8039594435c68eb4f780f3e9bf3972a7399c4b1",
```

Comment:
question nonblocking (this is a temporary hack): do we need rev?

Reply:
I don't think so, I basically copied the resolved input from the lockfile without caring too much since this is a hack.